src/core/meshcop/dataset_manager_ftd.cpp

```diff
@@ -274,5 +274,44 @@ void PendingDataset::HandleSet(Coap::Header &aHeader, Message &aMessage, const I
     return;
 }
 
+void PendingDataset::ApplyActiveDataset(const Timestamp &aTimestamp, Message &aMessage)
+{
+    uint16_t offset = aMessage.GetOffset();
+    DelayTimerTlv delayTimer;
+    uint8_t flags;
+
+    VerifyOrExit(mNetif.GetMle().IsAttached(), ;);
+
+    while (offset < aMessage.GetLength())
+    {
+        OT_TOOL_PACKED_BEGIN
+        struct
+        {
+            Tlv tlv;
+            uint8_t value[Dataset::kMaxValueSize];
+        } OT_TOOL_PACKED_END data;
+
+        aMessage.Read(offset, sizeof(Tlv), &data.tlv);
+        aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
+        mNetwork.Set(data.tlv);
+        offset += sizeof(Tlv) + data.tlv.GetLength();
+    }
+
+    // add delay timer tlv
+    delayTimer.Init();
+    delayTimer.SetDelayTimer(mNetif.GetLeader().GetDelayTimerMinimal());
+    mNetwork.Set(delayTimer);
+
+    // add pending timestamp tlv
+    mNetwork.SetTimestamp(aTimestamp);
+    HandleNetworkUpdate(flags);
+
+    // reset delay timer
+    ResetDelayTimer(kFlagNetworkUpdated);
+
+exit:
+    {}
```

Comment:
Instead of using an empty code block here, couldn't you just explicitly call `return;`?  I think that would be cleaner then `{}`.